### PR TITLE
fix: cap SKBUILD_PROJECT_VERSION to four components

### PR DESCRIPTION
**Pull request by henryiii**
:robot: _AI text below_ :robot:

Caps `SKBUILD_PROJECT_VERSION` to at most four dot-separated release components (`major.minor.patch.tweak`) so it is always valid for CMake's strict `project(VERSION ...)` parser. This addresses #1367, where projects currently hand-strip PEP 440 suffixes.

Previously `SKBUILD_PROJECT_VERSION` was release-only but not guaranteed CMake-valid: `version.release` can have more than four components (`(1,2,3,4,5)` -> invalid). It is now truncated via `version.release[:4]`.

No zero-padding is added. CMake accepts up to four non-negative integer segments and is happy with a bare one- or two-component version, so `2` and `2.6` are left as-is. The `release` segment is already free of epoch/pre/post/dev/local. The full, unmodified version remains available as `SKBUILD_PROJECT_VERSION_FULL`, so no new variable is needed.

Examples:
- `2` -> `2`
- `2.6` -> `2.6`
- `1.2.3.4` -> `1.2.3.4`
- `1.2.3.4.5` -> `1.2.3.4`
- epoch/pre/post/dev/local are already stripped by `version.release`, e.g. `1!2.3.4` -> `2.3.4`, `1.0.0a1` -> `1.0.0`, `1.2.3.dev4+local` -> `1.2.3`

### Compatibility note

This changes the existing variable rather than adding a new one. Only versions with 5+ release components now render differently in `SKBUILD_PROJECT_VERSION`. Anyone needing the exact value should use `SKBUILD_PROJECT_VERSION_FULL`.

## Test plan
- `tests/test_builder.py::test_builder_project_version_cmake` is parametrized over the edge cases, asserting both the capped `SKBUILD_PROJECT_VERSION` and the untouched `SKBUILD_PROJECT_VERSION_FULL` in the generated init cache.
- `prek -a --quiet` clean (ruff + mypy + typos).
- Docs: updated the `SKBUILD_PROJECT_VERSION` entry in `docs/guide/cmakelists.md`.

### CODE REVIEW

**Comment by henryiii** [comment]
I think we should probably just cap the regular version one to three digits instead. We have a full version for more.

File: src/scikit_build_core/builder/builder.py [hunk 1]
```diff
@@ -226,8 +226,11 @@ def configure(
             canonical_name = name.replace("-", "_").replace(".", "_")
             cache_config["SKBUILD_PROJECT_NAME"] = canonical_name
         if version is not None:
+            # Cap to four components so it is valid for project(VERSION ...),
+            # which accepts up to major.minor.patch.tweak. No padding is needed;
+            # SKBUILD_PROJECT_VERSION_FULL retains the full version.
```
Comment (henryiii) [inline src/scikit_build_core/builder/builder.py]:
```suggestion
            # Cap to four components so it is valid for project(VERSION ...)
```
